src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentEventResource.java

```diff
@@ -0,0 +1,90 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import static de.tum.in.www1.artemis.web.rest.util.ResponseUtil.forbidden;
+
+import java.net.URISyntaxException;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.analytics.TextAssessmentEvent;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.TextAssessmentEventRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+/**
+ * REST controller for managing TextAssessmentEventResource.
+ */
+@RestController
+@RequestMapping("/api/text-assessment-event")
+public class TextAssessmentEventResource {
+
+    private final Logger log = LoggerFactory.getLogger(TextAssessmentEventResource.class);
+
+    private final TextAssessmentEventRepository textAssessmentEventRepository;
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final UserRepository userRepository;
+
+    private final CourseRepository courseRepository;
+
+    public TextAssessmentEventResource(TextAssessmentEventRepository textAssessmentEventRepository, AuthorizationCheckService authCheckService, UserRepository userRepository,
+            CourseRepository courseRepository) {
+        this.textAssessmentEventRepository = textAssessmentEventRepository;
+        this.authCheckService = authCheckService;
+        this.userRepository = userRepository;
+        this.courseRepository = courseRepository;
+    }
+
+    /**
+     * This function retrieves all of the events from the 'text_assessment_event' table
+     * @return returns a List of TextAssessmentEvent's
+     */
+    @GetMapping("/get-events")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<List<TextAssessmentEvent>> findAllEvents() {
+        List<TextAssessmentEvent> events = textAssessmentEventRepository.findAll();
+        return ResponseEntity.ok().body(events);
+    }
```

Comment:
Either remove this or limit it to admin users.

Reply:
Ok, I'll set it to admin, can think this would be useful at some point.